src/core/inc/GslMatrix.h

```diff
@@ -197,6 +197,17 @@ class GslMatrix : public Matrix
   //! This function multiplies \c this matrix by vector \c x and returns the resulting vector.
   GslVector  multiply                  (const GslVector& x) const;
 
+  //! This function multiplies \c this matrix by vector \c x and fills
+  //the preallocated vector \c y.
+  void       multiply                  (const GslVector& x, GslVector& y) const;
+
+  //! This function multiplies \c this matrix by matrix \c X and returns the resulting matrix.
+  GslMatrix  multiply                  (const GslMatrix& X) const;
```

Comment:
This may be fine, but for high-dimensional problems this returns an expensive copy.  Other alternatives are to fill a preallocated `GslMatrix`, or a non-`const` method that just updates `*this`.